charmhub_lp_tools/revisions.py

```diff
@@ -0,0 +1,363 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Implement the print-revisions command."""
+
+import argparse
+import json
+import logging
+import os
+import pathlib
+import textwrap
+from typing import (
+    Dict,
+    List,
+    Set,
+    Union,
+)
+
+from prettytable import PrettyTable
+
+from .charm_project import (
+    CharmChannel,
+)
+
+from .group_config import GroupConfig
+from .parsers import (
+    parse_channel,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+def setup_parser(
+    subparser: argparse.ArgumentParser
+) -> argparse.ArgumentParser:
+    """Setup parser for print-revisions command."""
+    parser = subparser.add_parser(
+        'print-revisions',
+        help=('Print all the revisions for a base, arch, channel or any '
+              'combination thereof.'),
+    )
+    parser.add_argument(
+        '-s', '--channel',
+        dest='channel',
+        metavar='CHANNEL',
+        required=True,
+        help='The channel as track/risk to clean.',
+    )
+    parser.add_argument(
+        '-b', '--base',
+        dest='bases',
+        action='append',
+        type=str,
+        help='Restrict to a particular base. Can be repeated.')
+    parser.add_argument(
+        '-a', '--arch',
+        dest='arches',
+        action='append',
+        type=str,
+        help='Restrict to a particular arch. Can be repeated.')
+    parser.add_argument(
+        '--format',
+        dest='format',
+        # choices=('table', 'json', 'html'),
+        choices=('table', 'json', 'rst'),
+        default='table',
+        type=str.lower,
+        help=('The format to output the report in. default is "table"'))
+    parser.add_argument(
+        '--tab-size',
+        dest='tab_size',
+        default=5,
+        type=int,
+        help=('The number of tabs in a tab group for RST format. Ignored for'
+              'other formats.'))
+    parser.add_argument(
+        '-o', '--out',
+        dest='output',
+        help=('Write the output to a file.  Default is STDOUT'))
+
+    parser.set_defaults(func=print_revisions)
+    return parser
+
+
+def get_revisions(channel: CharmChannel,
+                  args_bases: Union[List[str], None],
+                  args_arches: Union[List[str], None],
+                  ) -> Dict[str, Dict[str, List[int]]]:
+    """Get the revisions by base -> arch -> [revisions].
+
+    :param channel: the charm channel to work against.
+    :param bases: Optional list of bases to restrict report to.
+    :param arches: Optional list of arches to restrict report to.
```

Comment:
the arguments don't match with the docstring

Reply:
Done